lib/default.nix

```diff
@@ -334,6 +334,20 @@ let
     */
     packages = toplevel: toplevel._packages;
 
+    /* Checks whether nixpkgs is recent enough for vmTools to support the customQemu argument.
+
+      This feature was added in https://github.com/NixOS/nixpkgs/commit/65c851cd7523c669b8fb25236b1c48283a2f43ec, so
+      technically the minimum version is 24.11.20240708.65c851c. However, versionAtLeast only compares versions
+      alphabetically, so the comparison's result will be incorrect for other commits made on the same day.
+
+      Instead, we compare against 24.11.20240709, which is the next day. This means this function returns false for
+      some commits that DO support the customQemu argument, but if it returns true, we can be 100% certain that this
+      is correct, and we can pass the customQemu argument to vmTools without an evaluation error.
+
+      vmToolsSupportsCustomQemu :: pkgs -> bool
+    */
+    vmToolsSupportsCustomQemu = pkgs: lib.versionAtLeast pkgs.lib.version "24.11.20240709";
```

Comment:
I think it's good enough to assume people are either on a release channel or on the *latest* unstable one, so no need to check for the date IMO.

And while I really appreciate the detailed description, I do think it's a bit verbose for an inline comment and the commit message might be a better place to explain how it came to be?

Reply:
> I think it's good enough to assume people are either on a release channel or on the latest unstable one, so no need to check for the date IMO.

I disagree. The entire nixpkgs revision history lasts forever. People might use outdated versions for example to bisect compatibility issues upstream, doing research or to get a specific version of a package. The `qemu` option is simply incompatible with a certain range of nixpkgs. I found what that range was, and this function encapsulates that knowledge as good as is feasible. I don't think I should actively write a more incorrect implementation of this check. What would be the benefit of that?

> And while I really appreciate the detailed description 

Thank you :)

>  I do think it's a bit verbose for an inline comment and the commit message might be a better place to explain how it came to be?

Why? The contribution experience for developers depends on good documentation. We're not generating formatted docs from these comments yet, but they are de-facto documentation and IMO should be of the quality that we would expect from documentation for developers.

Again, I don't understand what would be the benefit of shortening this comment? If you can think of a more terse formulation, I'm all for it, but the fact that this function even exists bears an explanation, and if I saw something like this, I would appreciate it being documented.

Reply:
> I disagree. The entire nixpkgs revision history lasts forever. People might use outdated versions for example to bisect compatibility issues upstream, doing research or to get a specific version of a package. The qemu option is simply incompatible with a certain range of nixpkgs.

Sure, but the current state of things is that diskos master is broken for the latest official release (without this PR) and we didn't catch that in CI or otherwise until @Nebucatnetzer ran into it at NixCon. So I don't think supporting each and every nixpkgs commit is a realistic goal and therefore would rather focus on actually supporting both -unstable and the latest stable release.
No big deal to keep the date in, I don't care too much about it as long as it's just a single check and not an if-else-tree of date-specific clauses ;)

> Again, I don't understand what would be the benefit of shortening this comment? 

I am not arguing against documentation :D the first and last line of that comment are totally fine. The 2 paragraphs in between would IMO be better placed in the commit message, because the do provide *historical* context. Comments should tell me what that function does and possible pitfalls. I think its good to keep documentation somewhat consistent and don't think we document the genesis of other values anywhere? (And neither do I think we should).

But again, no strong feelings. I'd rather have VMs on diskos master branch work again with nixos stable sooner than later and don't want to block anything here.

Reply:
> until @Nebucatnetzer ran into it at NixCon

oh no, was it during a live demo? 😬